js/src/setup-ads/ads-stepper/index.js

```diff
@@ -27,11 +29,42 @@ import {
  */
 const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const initHasAdsConnectionRef = useRef( null );
+
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasResolvedGoogleAdsAccount,
+		hasGoogleAdsConnection,
+	} = useGoogleAdsAccount();
+
+	const {
+		hasAccess,
+		hasFinishedResolution: hasResolvedAdsAccountStatus,
+		step: adsAccountSetupStep,
+	} = useGoogleAdsAccountStatus();
+
 	useEventPropertiesFilter( FILTER_ONBOARDING, {
 		context: CONTEXT_ADS_ONBOARDING,
 		step,
 	} );
 
+	if (
+		! hasResolvedGoogleAdsAccount ||
+		! hasResolvedAdsAccountStatus ||
+		googleAdsAccount === null
+	) {
+		return null;
+	}
```

Comment:
What if we also check the `initHasAdsConnectionRef` here too, so that we only return an empty page if we don't already know that the state of whether the connection page should be shown?

```suggestion
	if (
		initHasAdsConnectionRef.current === null &&
		! ( hasResolvedGoogleAdsAccount && hasResolvedAdsAccountStatus )
	) {
		return null;
	}
```

Reply:
Make sense, I have made this change (slightly different way though ). 


@joemcgill @ankitguptaindia we will still have delays if the Google ads account is not connected. I have replaced the blank screen with a spinner instead for better UX. 

https://github.com/user-attachments/assets/8bb7983c-9e07-4e07-86c4-93082acf82a6